frontend/app/services/work-package-service.js

```diff
@@ -75,6 +76,47 @@ module.exports = function($http,
   }
 
   var WorkPackageService = {
+    initializeWorkPackage: function(projectIdentifier, initialData) {
+      var changes = _.clone(initialData);
+      if (initialData.type) {
+        changes._links = changes.links || {};
+        changes._links.type = {
+          href: initialData.type
+        };
+        delete changes.type;
+      }
+      var wp = {
+        embedded: {},
+        props: {},
+        links: {
+          update: HALAPIResource
+            .setup(PathHelper
+              .projectWorkPackagesFormPath(projectIdentifier)),
+          updateImmediately: HALAPIResource.setupLink(
+            PathHelper.projectWorkPackagesPath(projectIdentifier),
+            { method: 'post' }
+          )
```

Comment:
Those two `HALAPIResource` calls look very suspicious:

* Which use do they have at all? They look like the `createWorkPackage` and `createWorkPackageImmediate` links in a project resource
    * they do **not** look like update links (mind the difference between update and create)
* The `setupLink` helper will prepend an `api/v3` prefix. I am not sure why the correct uri is not passed here in the first place
* From a callers perspective (reading those lines above) I can't tell why `update` needs a `setup` and `updateImmediately` needs a `setupLink`. The names look arbitrary